plugins/Core/edu.toronto.cs.se.mmint.operator/src/edu/toronto/cs/se/mmint/operator/slice/Slice.java

```diff
@@ -84,27 +87,19 @@ public Input(Map<String, Model> inputsByName) {
 		}
 	}
 
-	// Returns the complete list of model elements that may be impacted
-	// by the model elements included in the original slicing criterion.
-	public Set<EObject> getImpactedElements(ModelRel criterion) throws MMINTException {
-
+	// Returns the complete set of model elements that may be impacted
+	// by the input set of model elements.
+	// By default, only the input elements are assumed to be impacted.
+	public Set<EObject> getAllImpactedElements(Set<EObject> elemSet) {
+		return elemSet;
+	}
+	
+	// Returns the set of model elements that may be directly impacted
+	// by the input model element (including itself).
+	// By default, only the input elements are assumed to be impacted.
+	public Set<EObject> getDirectlyImpactedElements(EObject elem) throws MMINTException {
```

Comment:
This method is never invoked in this operator.
If you want it to be overridden, then you should use it, otherwise it does not define a common behavior of Slice.

Reply:
As far as I understand, you're switching from the previous recursive api to two apis that navigate the model in a non-recursive way.
The side effect of this though is that you're switching from a depth-first search to a breadth-first. Do you have some numbers or experiments to back it up?
More importantly though, if you don't pass the already-impacted elements to getDirectlyImpactedElements(), you're going to waste time collecting elements that may already have been impacted.